client/src/app/site/motions/components/statute-paragraph-list/statute-paragraph-list.component.ts

```diff
@@ -0,0 +1,181 @@
+    templateUrl: './statute-paragraph-list.component.html',
+    styleUrls: ['./statute-paragraph-list.component.scss']
+})
+export class StatuteParagraphListComponent extends BaseComponent implements OnInit {
+    /**
+     * content of the ellipsis menu
+     */
+    public menuList: EllipsisMenuItem[] = [
+        {
+            text: 'Sort statute paragraphs',
+            action: 'sortStatuteParagraphs'
+        }
+    ];
+
+    public statuteParagraphToCreate: StatuteParagraph | null;
+
+    /**
+     * Source of the Data
+     */
+    public statuteParagraphs: ViewStatuteParagraph[] = [];
+
+    /**
+     * The current focussed formgroup
+     */
+    public updateForm: FormGroup;
+
+    public createForm: FormGroup;
+
+    public openId: number | null;
+    public editId: number | null;
+
+    /**
+     * The usual component constructor
+     * @param titleService
+     * @param translate
+     * @param repo
+     * @param formBuilder
+     */
+    public constructor(
+        protected titleService: Title,
+        protected translate: TranslateService,
+        private repo: StatuteParagraphRepositoryService,
+        private formBuilder: FormBuilder,
+        private promptService: PromptService
+    ) {
+        super(titleService, translate);
+        const form = {
+            title: ['', Validators.required],
+            text: ['', Validators.required]
+        };
+        this.createForm = this.formBuilder.group(form);
+        this.updateForm = this.formBuilder.group(form);
+    }
+
+    /**
+     * Event on Key Down in update or create form. Do not provide the viewSection for the create form.
+     */
+    public keyDownFunction(event: KeyboardEvent, viewStatuteParagraph?: ViewStatuteParagraph): void {
+        if (event.keyCode === 13) {
+            if (viewStatuteParagraph) {
+                this.onSaveButton(viewStatuteParagraph);
+            } else {
+                this.create();
+            }
+        }
+    }
+
+    /**
+     * Init function.
+     *
+     * Sets the title and gets/observes statute paragrpahs from DataStore
+     */
+    public ngOnInit(): void {
+        super.setTitle('Statute paragraphs');
+        this.repo.getViewModelListObservable().subscribe(newViewStatuteParagraphs => {
+            this.statuteParagraphs = newViewStatuteParagraphs;
+        });
+    }
+
+    /**
+     * Add a new Section.
+     */
+    public onPlusButton(): void {
+        if (!this.statuteParagraphToCreate) {
+            this.statuteParagraphToCreate = new StatuteParagraph();
+            this.createForm.setValue({
+                title: '',
+                text: ''
+            });
+        }
+    }
+
+    public create(): void {
+        if (this.createForm.valid) {
+            this.statuteParagraphToCreate.patchValues(this.createForm.value as StatuteParagraph);
+            this.repo.create(this.statuteParagraphToCreate).subscribe(resp => {
+                this.statuteParagraphToCreate = null;
```

Comment:
could also reset the form here, or does this happen automatically?

Reply:
The form is closed here and resetted if opened by the user again.